extensions/ql-vscode/src/model-editor/model-editor-module.ts

```diff
@@ -125,17 +129,24 @@ export class ModelEditorModule extends DisposableObject {
       }
 
       const language = db.language;
-      if (
-        !SUPPORTED_LANGUAGES.includes(language) ||
-        !isQueryLanguage(language)
-      ) {
+
+      // Ruby is only enabled when the config setting is set
+      const isSupportedLanguage =
+        SUPPORTED_LANGUAGES.includes(language) ||
+        (language === QueryLanguage.Ruby && this.modelConfig.enableRuby);
+
+      if (!isSupportedLanguage || !isQueryLanguage(language)) {
         void showAndLogErrorMessage(
           this.app.logger,
           `The CodeQL Model Editor is not supported for ${language} databases.`,
         );
         return;
       }
 
+      const definition = getModelsAsDataLanguage(language);
+
+      const initialMode = definition.availableModes?.[0] ?? Mode.Application;
```

Comment:
```suggestion
      const initialMode = definition.availableModes?.[0] ?? INITIAL_MODE;
```
where `INITIAL_MODE` comes from `mode.ts`. Just so we reuse the constant instead of hard-coding `Mode.Application`.